pkg/gcv/result.go

```diff
@@ -128,18 +119,18 @@ func (r *Result) ToInsights() []*Insight {
 }
 
 func (r *Result) ToViolations() ([]*validator.Violation, error) {
-	ancestryPath, found, err := unstructured.NestedString(r.CAIResource, ancestryPathKey)
+	auxMetadata := map[string]interface{}{}
+	ancestryPath, found, err := unstructured.NestedString(r.InputResource, ancestryPathKey)
 	if err != nil {
-
-		return nil, errors.Wrapf(err, "error getting ancestry path from %v", r.CAIResource)
+		return nil, errors.Wrapf(err, "error getting ancestry path from %v", r.InputResource)
 	}
-	if !found {
-		return nil, errors.Errorf("ancestry path not found in %v", r.CAIResource)
+	if found {
```

Comment:
If it is not found, auxMetadata won't be set with the key. In that case, can the flow continue into the loop for ConstraintViolations and if so, it may end up setting empty metadata for the violation.

Reply:
This is actually already enforced for CAI resources as part of ReviewAsset - it runs ValidateAsset and SanitizeAncestryPath, which do basic validation / cleanup of the asset structure.

Reply:
I've added validation of the TF resource change structure in [326dd84](https://github.com/GoogleCloudPlatform/config-validator/pull/176/commits/326dd842c5fc4db8fc73efb84579c8a9acb8d234)